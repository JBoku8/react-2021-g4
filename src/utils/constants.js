export const AUTH_TOKEN = 'auth.token';
export const CREDIT_CARDS_KEY = 'CREDIT_CARDS_KEY';
