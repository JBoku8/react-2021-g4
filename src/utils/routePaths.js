export const COUNTER_PATH = '/counter';
export const NOTES_PATH = '/notes';
export const PROFILE_PATH = '/profile';
export const CREDIT_CARDS_PATH = '/credit-cards';
export const REDUX_COUNTER_PATH = '/redux-counter';
export const HOME_PATH = '/';

// auth
export const AUTH_PATH = '/auth';
export const SIGN_IN_PATH = '/auth/sign-in';
export const SIGN_UP_PATH = '/auth/sign-up';
