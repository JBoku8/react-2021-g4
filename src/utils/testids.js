export const TEST_IDS = {
  greeting: {
    title: 'greeting.title',
    description: 'greeting.description',
  },
};

export default TEST_IDS;
