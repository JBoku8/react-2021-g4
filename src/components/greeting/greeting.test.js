import { render } from '@testing-library/react';
import { TEST_IDS } from '../../utils/testids';

import Greeting from './greeting';

describe('Greeting component test', () => {
  it('should render with defaults', () => {
    const component = render(<Greeting />);
    const defaultTitle = component.getByTestId(TEST_IDS.greeting.title);
    const defaultDesc = component.getByTestId(TEST_IDS.greeting.description);

    expect(defaultTitle).toBeInTheDocument();
    expect(defaultTitle).toHaveTextContent('default title');
    expect(defaultTitle).toHaveAttribute('title', 'default title');

    expect(defaultDesc).toBeInTheDocument();
    expect(defaultDesc).toHaveAttribute('title', 'default description');
  });

  it('should render with specified props', () => {
    const mockProps = {
      title: 'mock-title',
      description: 'mock-description',
    };
    const component = render(<Greeting {...mockProps} />);
    const defaultTitle = component.getByTestId(TEST_IDS.greeting.title);
    const defaultDesc = component.getByTestId(TEST_IDS.greeting.description);

    expect(defaultTitle).toHaveTextContent(mockProps.title);
    expect(defaultDesc).toHaveTextContent(mockProps.description);
  });
});

describe('Greeting snapshots', () => {
  test('default snap 1', () => {
    const component = render(<Greeting />);
    expect(component).toMatchSnapshot();
  });

  test('default snap 2', () => {
    const mockProps = {
      title: 'mock-title',
      description: 'mock-description',
    };
    const component = render(<Greeting {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
});
