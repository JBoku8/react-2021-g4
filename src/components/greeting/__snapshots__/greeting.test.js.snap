// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting snapshots default snap 1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row greeting-container"
      >
        <h1
          class="text-success"
          data-testid="greeting.title"
          title="default title"
        >
          default title
        </h1>
        <h4
          class="text-muted"
          data-testid="greeting.description"
          title="default description"
        >
          default description
        </h4>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row greeting-container"
    >
      <h1
        class="text-success"
        data-testid="greeting.title"
        title="default title"
      >
        default title
      </h1>
      <h4
        class="text-muted"
        data-testid="greeting.description"
        title="default description"
      >
        default description
      </h4>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Greeting snapshots default snap 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row greeting-container"
      >
        <h1
          class="text-success"
          data-testid="greeting.title"
          title="mock-title"
        >
          mock-title
        </h1>
        <h4
          class="text-muted"
          data-testid="greeting.description"
          title="mock-description"
        >
          mock-description
        </h4>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row greeting-container"
    >
      <h1
        class="text-success"
        data-testid="greeting.title"
        title="mock-title"
      >
        mock-title
      </h1>
      <h4
        class="text-muted"
        data-testid="greeting.description"
        title="mock-description"
      >
        mock-description
      </h4>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
